core/trino-spi/src/main/java/io/trino/spi/block/DictionaryBlock.java

```diff
@@ -52,37 +52,25 @@
     private final DictionaryId dictionarySourceId;
     private final boolean mayHaveNull;
 
-    public DictionaryBlock(Block dictionary, int[] ids)
-    {
-        this(ids.length, dictionary, ids);
-    }
-
     public DictionaryBlock(int positionCount, Block dictionary, int[] ids)
     {
-        this(0, positionCount, dictionary, ids, false, randomDictionaryId());
-    }
-
-    public DictionaryBlock(int positionCount, Block dictionary, int[] ids, DictionaryId dictionaryId)
-    {
-        this(0, positionCount, dictionary, ids, false, dictionaryId);
+        this(0, positionCount, dictionary, ids, false, false, randomDictionaryId());
     }
 
-    public DictionaryBlock(int positionCount, Block dictionary, int[] ids, boolean dictionaryIsCompacted)
-    {
-        this(0, positionCount, dictionary, ids, dictionaryIsCompacted, randomDictionaryId());
-    }
-
-    public DictionaryBlock(int positionCount, Block dictionary, int[] ids, boolean dictionaryIsCompacted, DictionaryId dictionarySourceId)
+    /**
+     * This should not only be used when creating a projection of another dictionary block.
+     */
+    public DictionaryBlock(int positionCount, Block dictionary, int[] ids, DictionaryId dictionarySourceId)
     {
-        this(0, positionCount, dictionary, ids, dictionaryIsCompacted, dictionarySourceId);
+        this(0, positionCount, dictionary, ids, false, false, dictionarySourceId);
     }
 
-    public DictionaryBlock(int idsOffset, int positionCount, Block dictionary, int[] ids, boolean dictionaryIsCompacted, DictionaryId dictionarySourceId)
+    DictionaryBlock(int idsOffset, int positionCount, Block dictionary, int[] ids)
     {
-        this(idsOffset, positionCount, dictionary, ids, dictionaryIsCompacted, false, dictionarySourceId);
+        this(idsOffset, positionCount, dictionary, ids, false, false, randomDictionaryId());
     }
 
-    public DictionaryBlock(int idsOffset, int positionCount, Block dictionary, int[] ids, boolean dictionaryIsCompacted, boolean isSequentialIds, DictionaryId dictionarySourceId)
```

Comment:
This prevents implementing `getPositions` for related dictionary ids (that share same `dictionaryId`)